Start again (again)

BoogieMonster1O1 opened the pull request.

BoogieMonster1O1 committed:
Progress 2

 Changeo be committed:
	modified:   src/main/java/io/github/hydos/holotext/HoloText.java
	new file:   src/main/java/io/github/hydos/holotext/config/Config.java
	modified:   src/main/java/io/github/hydos/holotext/core/EntityHoloText.java
	modified:   src/main/java/io/github/hydos/holotext/core/IHoloTextAcces.java
	new file:   src/main/java/io/github/hydos/holotext/core/command/CommandHolo.java
	modified:   src/main/java/io/github/hydos/holotext/mixin/MixinMinecraftServer.java

BoogieMonster1O1 committed:
Forgot to serialize

 Changes to be committed:
	modified:   src/main/java/io/github/hydos/holotext/core/command/CommandHolo.java

BoogieMonster1O1 committed:
Removal

 Changes to be committed:
	modified:   src/main/java/io/github/hydos/holotext/core/command/CommandHolo.java

BoogieMonster1O1 committed:
Do stuff at mod init

 Changes to be committed:
	modified:   src/main/java/io/github/hydos/holotext/HoloText.java

hYdos commented on src/main/java/io/github/hydos/holotext/HoloText.java:
```diff
@@ -4,58 +4,75 @@
 	private static final Path CONFIG_PATH = FabricLoader.getInstance().getConfigDir().resolve("holotext.json5");
 	private static final Jankson JANKSON = Jankson.builder().build();
-	private static final Codec<List<Entry>> ENTRIES_CODEC = Codec.list(Entry.CODEC);
+	private static final Consumer<String> PRINT_TO_STDERR = System.err::println;
```
how lazy can you be

BoogieMonster1O1 replied:
is that badd

hYdos replied:
yes. fix or remove

BoogieMonster1O1 replied:
no

BoogieMonster1O1 replied:
i leik it

hYdos replied:
guess i wont merge :p

BoogieMonster1O1 replied:
concern. explain why

hYdos replied:
because thats more jank than jankson

hYdos replied:
and dfu

BoogieMonster1O1 replied:
it's just a consumer

hYdos commented on src/main/java/io/github/hydos/holotext/config/Config.java:
```diff
@@ -0,0 +1,23 @@
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+public class Config {
```
Codecs are the bane of my existence so i hope you got this right

hYdos commented on src/main/java/io/github/hydos/holotext/mixin/MixinMinecraftServer.java:
```diff
@@ -12,6 +13,6 @@
     @Inject(at = @At("RETURN"), method = "createCommandManager")
     public void intercept(CallbackInfoReturnable<CommandManager> infoIn) {
-
+        infoIn.getReturnValue().registerCommand(new CommandHolo());
```
we should just get commands-v1 working properly :p

BoogieMonster1O1 replied:
commands v1 doesn't add it to /help

hYdos replied:
make it do it

BoogieMonster1O1 replied:
concern
i don't think it's possible without breaking api

hYdos replied:
no one uses v1 yet

hYdos replied:
(properly)

BoogieMonster1O1 replied:
explain that to @shedaniel 
do you think anyone used those seperate registry events? no
did I still have to keep them? yes

hYdos replied:
huh

BoogieMonster1O1 committed:
SHADOWING WORKS

 Changes to be committed:
	modified:   build.gradle
	modified:   gradle.properties
	modified:   src/main/java/io/github/hydos/holotext/HoloText.java
	deleted:    src/main/java/io/github/hydos/holotext/config/CodecWorld.java
	deleted:    src/main/java/io/github/hydos/holotext/config/Config.java
	deleted:    src/main/java/io/github/hydos/holotext/config/Entry.java
	new file:   src/main/java/io/github/hydos/holotext/core/ArmorStandExtensions.java
	new file:   src/main/java/io/github/hydos/holotext/core/Config.java
	deleted:    src/main/java/io/github/hydos/holotext/core/EntityHoloText.java
	new file:   src/main/java/io/github/hydos/holotext/core/HoloTexts.java
	deleted:    src/main/java/io/github/hydos/holotext/core/IHoloTextAccess.java
	deleted:    src/main/java/io/github/hydos/holotext/core/IServerWorldAccess.java
	deleted:    src/main/java/io/github/hydos/holotext/core/ITextReader.java
	deleted:    src/main/java/io/github/hydos/holotext/core/ITextWriter.java
	deleted:    src/main/java/io/github/hydos/holotext/core/IVectorReader.java
	deleted:    src/main/java/io/github/hydos/holotext/core/IVectorWriter.java
	deleted:    src/main/java/io/github/hydos/holotext/core/command/CommandHolo.java
	deleted:    src/main/java/io/github/hydos/holotext/mixin/AccessorArmorStandEntity.java
	new file:   src/main/java/io/github/hydos/holotext/mixin/ArmorStandEntityMixin.java
	renamed:    src/main/java/io/github/hydos/holotext/mixin/MixinMinecraftServer.java -> src/main/java/io/github/hydos/holotext/mixin/MinecraftServerMixin.java
	modified:   src/main/resources/holotext.mixins.json

BoogieMonster1O1 commented:
now serializes the data directly to the holotext.
removed a bunch of classes too

hYdos reviewed (approved):
good

hYdos commented on gradle.properties:
```diff
@@ -4,7 +4,7 @@ org.gradle.jvmargs=-Xmx2G -XstartOnFirstThread
 	# check these on https://fabricmc.net/use
 	minecraft_version = 1.8.9
-	yarn_mappings = 1.8.9+build.202009070256
+	yarn_mappings = 1.8.9+build.202009230433
```
![image](https://user-images.githubusercontent.com/25501603/93980959-a9a70700-fdc2-11ea-95f1-fbff47157767.png)

BoogieMonster1O1 replied:
that will require replacing every single entity method in the mod with `method_xxxx`

hYdos replied:
lmao

BoogieMonster1O1 committed:
e:b:ic command

 Changes to be committed:
	modified:   src/main/java/io/github/hydos/holotext/HoloText.java
	modified:   src/main/java/io/github/hydos/holotext/core/Config.java
	new file:   src/main/java/io/github/hydos/holotext/core/HoloTextCommand.java
	modified:   src/main/java/io/github/hydos/holotext/core/HoloTexts.java
	modified:   src/main/java/io/github/hydos/holotext/mixin/MinecraftServerMixin.java

BoogieMonster1O1 committed:
Creation

 Changes to be committed:
	modified:   src/main/java/io/github/hydos/holotext/HoloText.java
	modified:   src/main/java/io/github/hydos/holotext/core/Config.java
	modified:   src/main/java/io/github/hydos/holotext/core/HoloTexts.java
